pkg/indexer.go

```diff
@@ -3,34 +3,46 @@ package pkg
 import (
 	"errors"
 	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 
 	"github.com/theredditbandit/pman/pkg/db"
 )
 
 const (
-	StatusBucket       = "projects"
-	ProjectPaths       = "projectPaths"
-	ProjectAliasBucket = "projectAliases"
+	StatusBucket       string = "projects"
+	ProjectPaths       string = "projectPaths"
+	ProjectAliasBucket string = "projectAliases"
+)
+
+var (
+	ErrDirname    error = errors.New("error providing a directory name")
+	ErrDirInvalid error = errors.New("error providing a valid directory name")
+	ErrIsNotDir   error = errors.New("error providing a file instead of a directory")
+	ErrIndexDir   error = errors.New("error indexing directory")
 )
 
 // InitDirs indexes a directory for project directories and writes the data to the DB
 func InitDirs(args []string) error {
 	// the file which identifies a project directory
 	projIdentifier := "README.md"
 	if len(args) != 1 {
-		return errors.New("Please provide a directory name")
+		fmt.Println("Please provide a directory name")
+		return ErrDirname
 	}
 	dirname := args[0]
 	if stat, err := os.Stat(dirname); os.IsNotExist(err) {
-		return fmt.Errorf("%s is not a directory", dirname)
+		fmt.Printf("%s is not a directory \n", dirname)
+		return ErrDirInvalid
 	} else if !stat.IsDir() {
-		return fmt.Errorf("%s is a file and not a directory", dirname)
+		fmt.Printf("%s is a file and not a directory \n", dirname)
+		return ErrIsNotDir
 	}
 	projDirs, err := indexDir(dirname, projIdentifier)
 	if err != nil {
-		return err
+		fmt.Println(err)
+		return ErrIndexDir
```

Comment:
No fmt.Println should be called.

Using this would be better

```go
return fmt.Errorf("%w: %w")
```

Or simply a join